### ISSUE TITLE
3.4.0 breaks compatibility with 3.3.0 around formulas.

### ISSUE THREAD

**Issue by sifear** [caxlsx/caxlsx#207]
3.4.0 does not escape formulas by default, while 3.3.0 does (if escape_formulas is not set either way).
Therefore sheets with formulas are now attempting to evaluate them, or the formulas will get removed altogether if invalid.

**Comment by kiskoza** [comment]
Hi. I created this snippet based on our [formula example](https://github.com/caxlsx/caxlsx/blob/master/examples/basic_formula_example.md) and ran with both `3.3.0` and `3.4.0`. After opening the generated files, I saw the formula calculated just as in our example in both files.

```ruby
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'
  gem 'caxlsx', ARGV[0]
end

require 'axlsx'

p = Axlsx::Package.new
wb = p.workbook

wb.add_worksheet(name: 'Basic Worksheet') do |sheet|
  sheet.add_row ['A', 'B', 'C', 'SUM']
  sheet.add_row [1, 2, 3, '=SUM(A2:C2)']
end

p.serialize "basic_formula_example_#{ARGV[0]}.xlsx"
```

Could you provide a short snippet to reproduce the issue?

**Comment by sifear** [comment]
I cannot reproduce whatever I do. It seems I am mistaken. The only change now I can see between 3.3 and 3.4 is that  3.4.0 includes an extra equal sign:
```
sheet.add_row %w[A B C SUM]
sheet.add_row [1, 2, 3, '@AAA']
sheet.add_row [1, 2, 3, '@BBB@BBBB']
sheet.add_row [1, 2, 3, '=CCC']
```
3.3.0 will generate: 
`@AAA` `@BBB@BBBB` and `=ccc` respectively

3.4.0 will generate
`=@aaa` `=@bbb@bbbb` and `=ccc` respectively

I don't know enough to tell if this is expected or not, it is just what we had issue in our system. We expect 3.3.0 output. But maybe that is wrong after all. 3.4.0 generated excel will open an error and will removes the formulas.

**Comment by tagliala** [comment]
It's probably the leading `@`, which now is a "formula" prefix and may have break something

https://github.com/caxlsx/caxlsx/blob/abd59ab74ac9c0e50e663a12866e8f8a6d579240/lib/axlsx/workbook/worksheet/cell.rb#L75

**Comment by johnnyshields** [comment]
Sorry for that. Fix is here: https://github.com/caxlsx/caxlsx/pull/209

**Comment by noniq** [comment]
Fixed in 3.4.1 (which has just been released). Thanks for your report!
